### PR TITLE
[v7r0] Fix and improve LocalCE and Slurm

### PR DESCRIPTION
**Pull request by aldbr**
This PR fixes some issues to better interact with the Slurm batch system.

BEGINRELEASENOTES
*Core
CHANGE: <batchplugin>TimeLeft modules become <batchplugin>ResourceUsage and inherit from an abstract module called ResourceUsage
NEW: TimeLeft/SLURMResourceUsage module to get resource usage from a SLURM installation
CHANGE: Move TimeLeft from Core/Utilities to Resources/Computing/BatchSystems
*Resources
FIX: LocalCE: proxy submission and getJobStatus
FIX: Slurm getJobStatus() making use of sacct
ENDRELEASENOTES

### CODE REVIEW

**Comment by fstagni** [comment]
A general thing: would anyone be against in moving the whole of `DIRAC.Core.Utilities.TimeLeft` into, e.g. `DIRAC.Resources.Computing.BatchSystems` ? I don't understand why this is in `Core`. @atsareg

**Comment by fstagni** [comment]
And one more things: all these per-batch system plugins (LSF, PBS, etc.) can (should?) IMHO inherit from the main `TimeLeft` class (which right now is in `DIRAC.Core.Utilities.TimeLeft.TimeLeft`)

**Comment by atsareg** [comment]
I agree with Federico that TimeLeft modules should all go from Core to DIRAC.Resources.Computing.BatchSystems

File: Resources/Computing/LocalComputingElement.py [hunk 1]
```diff
@@ -215,20 +220,29 @@ def getCEStatus(self):
...
-    stampList = []
+    resultDict = {}
+    jobDict = {}
+
+    # Extract the batch job ID from the full DIRAC job ID
     for job in jobIDList:
       stamp = os.path.basename(urlparse(job).path)
-      stampList.append(stamp)
+      jobDict[stamp] = job
+    stampList = jobDict.keys()
```
Comment (fstagni) [inline Resources/Computing/LocalComputingElement.py]: Please don't use `.keys()` (python 3...). You can use 
```suggestion
    stampList = list(jobDict)
```

**Comment by aldbr** [comment]
> And one more things: all these per-batch system plugins (LSF, PBS, etc.) can (should?) IMHO inherit from the main `TimeLeft` class (which right now is in `DIRAC.Core.Utilities.TimeLeft.TimeLeft`)

`TimeLeft` computes the time left from the resource usage of a given batch system. 
The batch system plugins only need to give their resource usage to `TimeLeft` so I just created a `ResourceUsage` base class that gather the plugins. Thus, all the batch system plugins inherit from this class and implement the method to give their resource usage.

**Comment by fstagni** [comment]
Closed-reopened to see the tests succeeding...

**Pull request merged**